### PR TITLE
Document all breaking changes and notable changes in PrestaShop V9

### PR DESCRIPTION
**Pull request by jolelievre**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | 9.x
| Description?  | Document all breaking changes and notable changes in PrestaShop V9
| Fixed ticket? | ~

### CODE REVIEW

File: modules/core-updates/9.0.md [hunk 1]
```diff
@@ -15,4 +15,250 @@ menuTitle: Changes in 9.0
...
+- [Remove compatibility for php versions inferior to PHP 8.1](https://github.com/PrestaShop/PrestaShop/pull/32514)
+- [Support for PHP 8.2](https://github.com/PrestaShop/PrestaShop/pull/29116)
+- [Support for PHP 8.3](https://github.com/PrestaShop/PrestaShop/pull/34485)
+
+## Core changes
+
+### Symfony Upgrade
+
+PrestaShop V9 has been upgraded to rely on Symfony 6.4, V8 was previously based on Symfony 4.4 so it's quite a big leap. As such the V9 includes all the breaking changes related to the Symfony Framework,
+you can see [more details about the changes in the core here](https://github.com/PrestaShop/PrestaShop/issues/33993) and if you can also check Symfony migration guides (see below).
```
Comment (nicosomb) [inline modules/core-updates/9.0.md]: no new line here

File: modules/core-updates/9.0.md [hunk 2]
```diff
@@ -15,4 +15,250 @@ menuTitle: Changes in 9.0
...
+- [UPGRADE FROM 5.3 to 5.4](https://github.com/symfony/symfony/blob/5.4/UPGRADE-5.4.md)
+- [UPGRADE FROM 5.X to 6.0](https://github.com/symfony/symfony/blob/6.4/UPGRADE-6.0.md)
+- [UPGRADE FROM 6.0 to 6.1](https://github.com/symfony/symfony/blob/6.4/UPGRADE-6.1.md)
+- [UPGRADE FROM 6.1 to 6.2](https://github.com/symfony/symfony/blob/6.4/UPGRADE-6.2.md)
+- [UPGRADE FROM 6.2 to 6.3](https://github.com/symfony/symfony/blob/6.4/UPGRADE-6.3.md)
+- [UPGRADE FROM 6.3 to 6.4](https://github.com/symfony/symfony/blob/6.4/UPGRADE-6.4.md)
+
+### Removed dependencies
+
+We followed Symfony recommendation and no longer use the `symfony/symfony` dependency but instead each sub package manually, the minimum version of PHP 8.1 also forced us to upgrade or removed some dependencies.
```
Comment (nicosomb) [inline modules/core-updates/9.0.md]:
```suggestion
We followed Symfony recommendations and no longer use the `symfony/symfony` dependency but instead each sub package manually, the minimum version of PHP 8.1 also forced us to upgrade or removed some dependencies.
```

File: modules/core-updates/9.0.md [hunk 3]
```diff
@@ -15,4 +15,250 @@ menuTitle: Changes in 9.0
...
+- `symfony/inflector`
+- `symfony/notifier`
+- `symfony/rate-limiter`
+- `symfony/semaphore`
+- `symfony/uid`
+- `symfony/workflow`
+
+### Upgraded dependencies
+
+Some dependencies are still present but were upgrade which comes with their own breaking changes, please refer to each dependency changelog to understand them in details if you depend on these dependencies:
```
Comment (nicosomb) [inline modules/core-updates/9.0.md]:
```suggestion
Some dependencies are still present but were upgraded which comes with their own breaking changes, please refer to each dependency changelog to understand them in details if you depend on these dependencies:
```

File: modules/core-updates/9.0.md [hunk 4]
```diff
@@ -15,4 +15,250 @@ menuTitle: Changes in 9.0
...
+
+The migration of pages imply several breaking changes, we won't detail all of them for each page but here is a summary:
+- the legacy URL is no longer reachable, you are automatically redirected to the migrated page
+- the legacy controller is removed along with its associated smarty templates
+- the legacy hooks related to controller workflow are no longer called
+- the page no longer uses smarty but twig
+
+Here is the list of migrated pages in v9:
+
+- Internation > Locations > States
```
Comment (nicosomb) [inline modules/core-updates/9.0.md]:
```suggestion
- International > Locations > States
```

File: modules/core-updates/9.0.md [hunk 5]
```diff
@@ -15,4 +15,250 @@ menuTitle: Changes in 9.0
...
+- Sell > Catalog > Attributes
+- Sell > Catalog > Feature
+- Sell > Catalog > Products (new version of the page introduce in PrestaShop V8 is now the only available one)
+  - [edition of features use a new optimized data structure in the form](https://github.com/PrestaShop/PrestaShop/pull/34763)
+
+## Breaking Changes
+
+### `trans` method
+
+Following [PrestaShop/PrestaShop#30415](https://github.com/PrestaShop/PrestaShop/pull/30415) the function trans() does NOT escape anymore strings. In v8 you could pass parameters like `htmspecialcharacters` or `addslashes` to `trans()` to perform additionnal escape, but it’s been removed.
```
Comment (nicosomb) [inline modules/core-updates/9.0.md]:
```suggestion
Following [PrestaShop/PrestaShop#30415](https://github.com/PrestaShop/PrestaShop/pull/30415) the function `trans()` does NOT escape anymore strings. In v8 you could pass parameters like `htmspecialcharacters` or `addslashes` to `trans()` to perform additionnal escape, but it’s been removed.
```

**Review by tleon — APPROVED** [review]

**Review by matthieu-rolland — APPROVED** [review]

**Comment by matks** [comment]
So fast 😁 I opened the PR this morning to review it and it's already merged, bravo guys

**Pull request merged**